sdk/tables/Azure.Data.Tables/src/TableClient.cs

```diff
@@ -780,16 +784,15 @@ public virtual async Task<Response<T>> GetEntityIfExistsAsync<T>(string partitio
                     selectArg,
                     filter: null,
                     context);
-                if (!response.IsError)
+                if (response.Status == (int)HttpStatusCode.NotFound)
                 {
-                    var dictionary = SerializationHelpers.ResponseToDictionary(response);
-                    var result = dictionary.ToTableEntity<T>();
-                    return Response.FromValue(result, response);
+                    return new NoBodyResponse<T>(response);
```

Comment:
I would have expected to see us return an internal type that derived directly from NullableResponse<T> here?  It feels awkward that we return a type that hides and then reinstates "HasValue".

Reply:
I'm not sure it's worth adding a new type unless the API behavior would be different. Is there something that adding a dedicated type here would add that I'm missing?

Reply:
It's what it would subtract.  I could cast this to Response<T> or ask `response is Response<T>` and reasonably assume I'd gotten a value for T, since that's what the semantics of Response<T> imply, and it hides `HasValue`.  If we're going to return a Response<T> when T is null, it feels like we're undoing all the work we've done to add a type that indicates the T could be null.  But I'm fine if you wanted to get an architect perspective to see if I'm being too nit-picky.

Reply:
While it seems like an edge case, it's a good point that someone could cast this to `Response<T>` and get bad hints. 

Would you name the internal type `NullResponse<T>`, `EmptyResponse<T>`, `NoValueResponse<T>`, (any other ideas)?

Reply:
What about `NoContentResponse<T>` - that would seem to align with the official nomenclature for an HTTP 204

Reply:
After introducing this new internal response type, I think `NoBodyResponse<T>` should effectively be deprecated moving forward. 

Public APIs that already return it are stuck (it's only a hand full of APIs that use it today), but going forward, it would have the same problem as called out here.

Reply:
Cool.  I was going to vote for `NullResponse<T>`, to match the corresponding `ValueResponse<T>` and parent, but I don't feel super strongly since it's an internal API.

Reply:
Seems like we are just trying to represent a NullableResponse that is, in fact, null. So NullResponse seems to convey that. We should probably seal this type so nobody tries to change the semantics.

Reply:
My current vote is `NoValueResponse<T>` since "nullable" is a misnomer we only used due to the similarity with the Nullable<T> API. But in implementation, what we are needing to do is create counterpart to `ValueResponse<T>`

Reply:
That makes sense to me.